sdk/core/src/services/syft/mod.rs

```diff
@@ -196,23 +263,212 @@ pub enum Error {
     /// Raised when there is an error while attempting to syft a Repository
     #[error("syft error: {0}")]
     SyftError(String),
+
+    /// Transparently handle error when actually executing Syft
+    #[error(transparent)]
+    SyftExecutionError(#[from] serde_json::error::Error),
```

Comment:
It feels like we could merge these two, and refactor now towards your transparent error improvement. I suspect the changes will only affect this file. Do you think now is the right time?  Also, can we dry this up? I have been standardizing on not having the `Error` suffix on enum variants. Does that seem right to you?

Reply:
Thank you for this, it caused a fantastic refactor.